heaps/min_heap.py

```diff
@@ -19,19 +23,32 @@ def __init__(self):
     def add(self, key, value = None):
         """ This method adds a HeapNode instance to the heap
             If value == None the new node's value should be set to key
-            Time Complexity: ?
-            Space Complexity: ?
+            Time Complexity: O(1) if adding to rear, else O(logn)
+            Space Complexity: O(1)
         """
-        pass
+        if value == None:
+            value = key
+        
+        node = HeapNode(key, value)
+        self.store.append(node)
+        self.heap_up(len(self.store) - 1)
 
     def remove(self):
```

Comment:
✨ however, space complexity is O(logn) due to `heap_down`'s recursive call stack